code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCard.java

```diff
@@ -0,0 +1,219 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.messaging;
+
+import com.adobe.marketing.mobile.MessagingEdgeEventType;
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.util.StringUtils;
+
+/**
+ * A {@link ContentCard} object encapsulates the information necessary for a non-disruptive yet
+ * interactive offer. Customers can use the Messaging SDK to render the content card in a
+ * pre-defined format or implement their own rendering.
+ */
+public class ContentCard {
+    private static final String SELF_TAG = "ContentCard";
+
+    // Plain-text title for the content card
+    private String title;
+    // Plain-text body representing the content for the content card
+    private String body;
+    // String representing a URI that contains an image to be used for this content card
+    private String imageUrl;
+    // Contains a URL to be opened if the user interacts with the content card
+    private String actionUrl;
+    // Required if actionUrl is provided. Text to be used in title of button or link in content card
+    private String actionTitle;
+    // Reference to parent ContentCardSchemaData instance
+    ContentCardSchemaData parent;
```

Comment:
Is it ok to hold a `SoftReference` or `WeakReference` here to allow the parent to be garage collected if there are no other external references expect this one?

Reply:
this variable was previously a `SoftReference`. was it changed to a hard reference for a reason?

Reply:
it needs a hard reference now to ensure tracking capabilities

Reply:
In that case, the proposition would still be in the in-memory map. Wondering if having a strong reference here would prevent the parent class from being garbage collected when it is replaced in the map and is no longer valid?